app/Jobs/ImportCSV.php

```diff
@@ -46,16 +46,63 @@ public function handle(CSVImportReader $reader)
         $filepath = Storage::disk('local')
             ->path('mismatch-files/' . $this->meta->filename);
 
-        DB::transaction(function () use ($reader, $filepath) {
-            $reader->lines($filepath)->each(function ($mismatchLine) {
-                $mismatch = Mismatch::make($mismatchLine);
-                if ($mismatch->type == null) {
-                    $mismatch->type = 'statement';
+        $mismatch_attrs = (new Mismatch())->getFillable();
+
+        DB::transaction(function () use ($reader, $filepath, $mismatch_attrs) {
+            $new_mismatches = [];
+            $where_clauses = [];
+
+            $mismatches_per_upload_user = DB::table('mismatches')
+                ->select($mismatch_attrs)
+                ->join('import_meta', 'mismatches.import_id', '=', 'import_meta.id')
+                ->where('import_meta.user_id', '=', $this->meta->user->id);
+
+            $reader->lines($filepath)->each(function ($mismatchLine) use (
+                &$new_mismatches,
+                &$where_clauses
+            ) {
+
+                $new_mismatch = Mismatch::make($mismatchLine);
+                $new_mismatches[] = $new_mismatch;
+                $mismatch_attributes = collect($new_mismatch->getAttributes());
+                $mismatch_attributes->forget('review_status');
+                $where_clause = [['review_status', '!=', 'pending']];
+                $mismatch_attributes->map(function ($attribute, $key) use (&$where_clause) {
+                    // key can be empty in the file but in the db always has statement by default
+                    if ($key != 'type' || $attribute != null) {
```

Comment:
I'm not sure why, but leaving the type out makes mismatches with an empty `type` column get reuploaded as duplicates. I think we want to avoid that

Reply:
Yes, that's why this was added. Because the type column will be converted to statement automatically when empty. So we don't want to take this column from the CSV into account.

Reply:
Yes, that's clear. What happens in practice, is that (at least locally for me) the one mismatch in the example file that has no type keeps getting reuploaded, although it has been reviewed. 
We should also consider the (improbable yet not impossible) case where the same mismatch is uploaded but with a qualifier type, but we leave out the type so we will consider it a duplicate, which would be wrong.

Reply:
@chukarave this shouldn't be the case, I will try this on my local as well